### ISSUE TITLE
problem with new toolchain `foss/2018.08` and `--try-toolchain` option

### ISSUE THREAD

**Issue by pescobar** [easybuilders/easybuild-framework#2615]
Hi 

I have installed the new toolchain `foss/2018.08` (still in the devel branch)
https://github.com/easybuilders/easybuild-easyconfigs/pull/6992

After installing the new toolchain the `try-toolchain` option doesn't work in combitnation with it e.g

This works fine using the common toolchains provided by easybuild:
`eb Python-2.7.14-foss-2018a.eb --try-toolchain-version=2018b --robot`

This crashes (the only change is version=2018b to version=2018.08):

```
eb Python-2.7.14-foss-2018a.eb --try-toolchain-version=2018.08 --robot
== temporary log file in case of crash /scratch/eb-aZ31sN/easybuild-a9TnOl.log
ERROR: Could not find easyconfig for foss toolchain version 2018.08
```

If I try to pass as argument to `--robot` the path of my local clone that includes the easyconfig for `foss/2018.08` then:

```
eb Python-2.7.14-foss-2018a.eb --try-toolchain-version=2018.08 --robot -r easyconfigs/easybuild/easyconfigs/
== temporary log file in case of crash /scratch/eb-I3rh9M/easybuild-eIbvaZ.log
ERROR: Target hierarchy {'lapack_family': 'OpenBLAS', 'name': 'foss', 'blas_family': 'OpenBLAS', 'version': '2018.08', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': 'OpenMPI'} should have binutils using GCCcore, can't determine mapping!
```

@boegel suggested that this could be a problem with the new toolchain

@ocaisa any idea what could be the problem ?

I used latest easybuild 3.7.0 for the tests.

**Comment by ocaisa** [comment]
I can't reproduce this with the current devel branches, works fine. I suspect it might be something in your local configuration that hasn't been taken into account, anything helpful in the debug logs?

Relevant error comes from [here](https://github.com/easybuilders/easybuild-framework/blob/26d6b8789049d25df93f95390fb6a049b7ad41be/easybuild/framework/easyconfig/tweak.py#L792). If you do a dry run on the toolchain easyconfig for `2018.08` does binutils appear in the result (we currently rely on this)?

**Comment by ocaisa** [comment]
Does `easyconfigs/easybuild/easyconfigs/` include all of the easyconfigs required to build the toolchain (and in particular the required `binutils/2.31.1`)?

**Comment by pescobar** [comment]
Thanks for your help @ocaisa 

My local clone includes the files. I have downloaded my branch including the toolchain easyconfigs like this:

`$> git clone https://github.com/pescobar/easybuild-easyconfigs.git -b foss.2018.08 ./to_delete`

Then I load the `EasyBuild/3.7.0` module and I try:

```
$> eb to_delete/easybuild/easyconfigs/f/foss/foss-2018.08.eb -r to_delete/easybuild/easyconfigs/ -D | grep binutils
 * [ ] $CFGS/b/binutils/binutils-2.31.1.eb (module: binutils/2.31.1)
 * [ ] $CFGS/b/binutils/binutils-2.31.1-GCCcore-8.2.0.eb (module: binutils/2.31.1-GCCcore-8.2.0)
```

If I try `--hide-deps=binutils` then the binutils installed centrally are found:

```
$> $ eb to_delete/easybuild/easyconfigs/f/foss/foss-2018.08.eb -r to_delete/easybuild/easyconfigs/ --hide-deps=binutils -D | grep binutils
 * [x] $CFGS/b/binutils/binutils-2.31.1.eb (module: binutils/.2.31.1)
 * [x] $CFGS/b/binutils/binutils-2.31.1-GCCcore-8.2.0.eb (module: binutils/.2.31.1-GCCcore-8.2.0)
```

 but if I try the installation it crashes, no matter if I use `--hidden-deps` or not:

```
$> eb to_delete/easybuild/easyconfigs/p/Python/Python-2.7.15-foss-2018b.eb -r to_delete/easybuild/easyconfigs/ --hide-deps=binutils --try-toolchain-version=2018.08 -D
== temporary log file in case of crash /scratch/eb-cvZcNZ/easybuild-fwr3Fk.log
ERROR: Target hierarchy {'lapack_family': 'OpenBLAS', 'name': 'foss', 'blas_family': 'OpenBLAS', 'version': '2018.08', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': 'OpenMPI'} should have binutils using GCCcore, can't determine mapping!

```

I am using the default easybuild options:

```
$ type eb
eb is aliased to `eb --modules-tool=Lmod'

$ eb --show-config
#
# Current EasyBuild configuration
# (C: command line argument, D: default value, E: environment variable, F: configuration file)
#
buildpath      (D) = /scicore/home/scicore/easybuild/.local/easybuild/build
containerpath  (D) = /scicore/home/scicore/easybuild/.local/easybuild/containers
installpath    (D) = /scicore/home/scicore/easybuild/.local/easybuild
repositorypath (D) = /scicore/home/scicore/easybuild/.local/easybuild/ebfiles_repo
robot-paths    (D) = /scicore/soft/apps/EasyBuild/3.7.0/lib/python2.7/site-packages/easybuild_easyconfigs-3.7.0-py2.7.egg/easybuild/easyconfigs
sourcepath     (D) = /scicore/home/scicore/easybuild/.local/easybuild/sources
```

Could this be related to binutils being hidden by `--hidden-deps` in my cluster?

**Comment by pescobar** [comment]
I have tried with a different cluster account and modifying the MODULEPATH so I don't see any of the installed modules in my cluster. Still it doesn't work. I am out of ideas.

I have uploaded a debug log in case anyone can add any light on this
https://gist.github.com/pescobar/54bbfff71079cc37320162db510d8d11

My last test was:
```
login20 tmp]$ echo $MODULEPATH 
/scicore/soft/modules/tools

login20 tmp]$ ml EasyBuild

login20 tmp]$ eb eb_develop/easybuild/easyconfigs/p/Python/Python-2.7.15-foss-2018b.eb -r eb_develop/easybuild/easyconfigs/ --try-toolchain=foss,2018.08 -D
== temporary log file in case of crash /scratch/eb-Aea7aO/easybuild-yphlYE.log
ERROR: Target hierarchy {'lapack_family': 'OpenBLAS', 'name': 'foss', 'blas_family': 'OpenBLAS', 'version': '2018.08', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': 'OpenMPI'} should have binutils using GCCcore, can't determine mapping!
```

**Comment by pescobar** [comment]
I tried to download latest develop version and unset my MODULEPATH. Still the same...totally puzzled...

```
$> echo $MODULEPATH 
/scratch/escobar/eb_develop/modules

$> eb --version
This is EasyBuild 3.7.1.dev0-ra59e3fab492501f02bd56d5ccbba72e29440c6bd (framework: 3.7.1.dev0-ra59e3fab492501f02bd56d5ccbba72e29440c6bd, easyblocks: 3.7.1.dev0-r04ca741fd825932b439f1a2d9d49879f91b49b5a) on host login20.cluster.bc2.ch.

$> eb Python-2.7.15-foss-2018b.eb -r --try-toolchain=foss,2018.08 -D
== temporary log file in case of crash /scratch/eb-wlb4XC/easybuild-ZQKItF.log
ERROR: Target hierarchy {'lapack_family': 'OpenBLAS', 'name': 'foss', 'blas_family': 'OpenBLAS', 'version': '2018.08', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': 'OpenMPI'} should have binutils using GCCcore, can't determine mapping!
```

**Comment by boegel** [comment]
@pescobar There's a bit of a lack of debug logging in that part of the code, but I think the problem is that somehow you don't have a `binutils` that's built with `GCCcore` anywhere in the dependency tree for `foss-2018.08.eb`.

I get this:

```
$ eb foss-2018.08.eb -D | grep binutils
 * [ ] $CFGS/b/binutils/binutils-2.31.1.eb (module: binutils/2.31.1)
 * [ ] $CFGS/b/binutils/binutils-2.31.1-GCCcore-8.2.0.eb (module: binutils/2.31.1-GCCcore-8.2.0)
```

Do you get a different result somehow?

**Comment by boegel** [comment]
@pescobar Also, can you provide a new debug log after applying the following patch?

**Comment by boegel** [comment]
```diff
diff --git a/easybuild/framework/easyconfig/tweak.py b/easybuild/framework/easyconfig/tweak.py
index 8b3a0187e..f4033453a 100644
--- a/easybuild/framework/easyconfig/tweak.py
+++ b/easybuild/framework/easyconfig/tweak.py
@@ -769,7 +769,10 @@ def get_dep_tree_of_toolchain(toolchain_spec, modtool):
                              toolchain_spec['name'], toolchain_spec['version'])
     ec = process_easyconfig(path, validate=False)
 
-    return [dep['ec'] for dep in resolve_dependencies(ec, modtool)]
+    res = [dep['ec'] for dep in resolve_dependencies(ec, modtool)]
+    _log.debug("Dependency tree for toolchain %s: %s", toolchain_spec, res)
+
+    return res
 
 
 def map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool):
```

**Comment by zao** [comment]
Getting the same as @pescobar on my Bionic machine, using a EasyBuild-develop.
Debug log: https://gist.github.com/zao/7b5cacd3294bdfa7e2c116fb59278a64

```
zao@freja:~$ ml EasyBuild-develop

zao@freja:~$ echo $MODULEPATH
/home/zao/.local/eb-develop/modules:/opt/Lmod-7.8.6/modulefiles/Linux:/opt/Lmod-7.8.6/modulefiles/Core:/opt/Lmod-7.8.6/lmod/lmod/modulefiles/Core

zao@freja:~$ eb --version
This is EasyBuild 3.7.1.dev0-ra59e3fab492501f02bd56d5ccbba72e29440c6bd (framework: 3.7.1.dev0-ra59e3fab492501f02bd56d5ccbba72e29440c6bd, easyblocks: 3.7.1.dev0-r19499dffcf97232a26cc0243dbf0b3ebd4a1716c) on host freja.

zao@freja:~$ eb Python-2.7.15-foss-2018b.eb -r --try-toolchain=foss,2018.08 -D
== temporary log file in case of crash /tmp/eb-VlOuka/easybuild-7efJmy.log
ERROR: Target hierarchy {'lapack_family': 'OpenBLAS', 'name': 'foss', 'blas_family': 'OpenBLAS', 'version': '2018.08', 'cuda': None, 'comp_family': 'GCC', 'mpi_family': 'OpenMPI'} should have binutils using GCCcore, can't determine mapping!
```

**Comment by boegel** [comment]
OK, now it's officially weird...

@zao Do you mind trying with after applying the small patch that should give some more debug info?

**Comment by zao** [comment]
In my configuration, all the dry-run configs are marked `[X]` after installing foss/2018.08.
The subset asked for:
```
zao@freja:~$ eb foss-2018.08.eb -D | grep binutils
 * [x] $CFGS/b/binutils/binutils-2.31.1.eb (module: binutils/2.31.1)
 * [x] $CFGS/b/binutils/binutils-2.31.1-GCCcore-8.2.0.eb (module: binutils/2.31.1-GCCcore-8.2.0)
```

**Comment by boegel** [comment]
The extra debug logging wasn't exactly what I was hoping for, but it was enough to find the culprit...

@zao @pescobar Please test the patch in #2617

**Comment by zao** [comment]
```
zao@freja:~/.local/eb-develop/easybuild-framework$ eb Python-2.7.15-foss-2018b.eb -r --try-toolchain=foss,2018.08 -D --debug
== temporary log file in case of crash /tmp/eb-JANdvG/easybuild-OsySOQ.log
Dry run: printing build status of easyconfigs and dependencies
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/m/M4/M4-1.4.18.eb (module: M4/1.4.18)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/b/Bison/Bison-3.0.5.eb (module: Bison/3.0.5)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/z/zlib/zlib-1.2.11.eb (module: zlib/1.2.11)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/h/help2man/help2man-1.47.4.eb (module: help2man/1.47.4)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/m/M4/M4-1.4.17.eb (module: M4/1.4.17)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/b/Bison/Bison-3.0.4.eb (module: Bison/3.0.4)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/f/flex/flex-2.6.4.eb (module: flex/2.6.4)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/b/binutils/binutils-2.31.1.eb (module: binutils/2.31.1)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/g/GCCcore/GCCcore-8.2.0.eb (module: GCCcore/8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/m/M4/M4-1.4.18-GCCcore-8.2.0.eb (module: M4/1.4.18-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/h/help2man/help2man-1.47.7-GCCcore-8.2.0.eb (module: help2man/1.47.7-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/z/zlib/zlib-1.2.11-GCCcore-8.2.0.eb (module: zlib/1.2.11-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/b/Bison/Bison-3.0.5-GCCcore-8.2.0.eb (module: Bison/3.0.5-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/f/flex/flex-2.6.4-GCCcore-8.2.0.eb (module: flex/2.6.4-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/b/binutils/binutils-2.31.1-GCCcore-8.2.0.eb (module: binutils/2.31.1-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/ncurses-6.1-GCCcore-8.2.0.eb (module: ncurses/6.1-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/Tcl-8.6.8-GCCcore-8.2.0.eb (module: Tcl/8.6.8-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/bzip2-1.0.6-GCCcore-8.2.0.eb (module: bzip2/1.0.6-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/libreadline-7.0-GCCcore-8.2.0.eb (module: libreadline/7.0-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/SQLite-3.24.0-GCCcore-8.2.0.eb (module: SQLite/3.24.0-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/libffi-3.2.1-GCCcore-8.2.0.eb (module: libffi/3.2.1-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/a/Autoconf/Autoconf-2.69-GCCcore-8.2.0.eb (module: Autoconf/2.69-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/a/Automake/Automake-1.16.1-GCCcore-8.2.0.eb (module: Automake/1.16.1-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/g/GCC/GCC-8.2.0-2.31.1.eb (module: GCC/8.2.0-2.31.1)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/l/libtool/libtool-2.4.6-GCCcore-8.2.0.eb (module: libtool/2.4.6-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/a/Autotools/Autotools-20180311-GCCcore-8.2.0.eb (module: Autotools/20180311-GCCcore-8.2.0)
 * [ ] /tmp/eb-JANdvG/tweaked_dep_easyconfigs/GMP-6.1.2-GCCcore-8.2.0.eb (module: GMP/6.1.2-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenBLAS/OpenBLAS-0.3.3-GCC-8.2.0-2.31.1.eb (module: OpenBLAS/0.3.3-GCC-8.2.0-2.31.1)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/n/numactl/numactl-2.0.12-GCCcore-8.2.0.eb (module: numactl/2.0.12-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/x/xorg-macros/xorg-macros-1.19.2-GCCcore-8.2.0.eb (module: xorg-macros/1.19.2-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/l/libpciaccess/libpciaccess-0.14-GCCcore-8.2.0.eb (module: libpciaccess/0.14-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/n/ncurses/ncurses-6.0.eb (module: ncurses/6.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/g/gettext/gettext-0.19.8.1.eb (module: gettext/0.19.8.1)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/x/XZ/XZ-5.2.4-GCCcore-8.2.0.eb (module: XZ/5.2.4-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/l/libxml2/libxml2-2.9.8-GCCcore-8.2.0.eb (module: libxml2/2.9.8-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/h/hwloc/hwloc-1.11.11-GCCcore-8.2.0.eb (module: hwloc/1.11.11-GCCcore-8.2.0)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/o/OpenMPI/OpenMPI-3.1.2-GCC-8.2.0-2.31.1.eb (module: OpenMPI/3.1.2-GCC-8.2.0-2.31.1)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/g/gompi/gompi-2018.08.eb (module: gompi/2018.08)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/f/FFTW/FFTW-3.3.8-gompi-2018.08.eb (module: FFTW/3.3.8-gompi-2018.08)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/s/ScaLAPACK/ScaLAPACK-2.0.2-gompi-2018.08-OpenBLAS-0.3.3.eb (module: ScaLAPACK/2.0.2-gompi-2018.08-OpenBLAS-0.3.3)
 * [x] /home/zao/.local/eb-develop/easybuild-easyconfigs/easybuild/easyconfigs/f/foss/foss-2018.08.eb (module: foss/2018.08)
 * [ ] /tmp/eb-JANdvG/tweaked_easyconfigs/Python-2.7.15-foss-2018.08.eb (module: Python/2.7.15-foss-2018.08)
== Temporary log file(s) /tmp/eb-JANdvG/easybuild-OsySOQ.log* have been removed.
== Temporary directory /tmp/eb-JANdvG has been removed.
zao@freja:~/.local/eb-develop/easybuild-framework$
```

**Comment by boegel** [comment]
To clarify: the problem only occurred when some of the modules (in particular the `binutils` ones) were already installed... The `retain_deps=True` that was added in #2617 to `resolve_dependencies` makes sure that it always gets all dependencies, regardless of existing modules.

**Comment by zao** [comment]
Build went fine until it was going to do the Python stuff, appears that `%(namelower)s` isn't expanded?
```
ERROR: Build of /tmp/eb-xeEGWZ/tweaked_easyconfigs/Python-2.7.15-foss-2018.08.eb failed (err: "build failed (first 300 chars): Couldn't find file setuptools-39.2.0.zip anywhere, and downloading it didn't work either... Paths attempted (in order): /tmp/eb-xeEGWZ/tweaked_easyconfigs/p/Python/extensions/setuptools-39.2.0.zip, /tmp/eb-xeEGWZ/tweaked_easyconfigs/p/Python/packages/setuptools-39.2.0.zip, /tmp/eb-xeEGWZ/tweaked_eas")
```
```
== 2018-10-12 20:58:17,199 filetools.py:1212 DEBUG Not creating existing path /home/zao/.local/eb-develop/sources/p/Python
== 2018-10-12 20:58:17,199 easyblock.py:728 DEBUG Trying to download file setuptools-39.2.0.zip from https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip ...
== 2018-10-12 20:58:17,199 filetools.py:484 DEBUG Trying to download setuptools-39.2.0.zip from https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip
== 2018-10-12 20:58:17,199 filetools.py:491 DEBUG Using timeout of 10 seconds for initiating download
== 2018-10-12 20:58:17,199 filetools.py:1184 INFO Creating directory /home/zao/.local/eb-develop/sources/p/Python/extensions (parents: True, set_gid: False, sticky: False)
== 2018-10-12 20:58:17,257 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -
2] Name or service not known>
== 2018-10-12 20:58:17,258 filetools.py:549 INFO Attempt 1 of downloading https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, trying again...
== 2018-10-12 20:58:17,259 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -2] Name or service not known>
== 2018-10-12 20:58:17,259 filetools.py:549 INFO Attempt 2 of downloading https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, trying again...
== 2018-10-12 20:58:17,260 filetools.py:535 WARNING IOError occurred while trying to download https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip: <urlopen error [Errno -2] Name or service not known>
== 2018-10-12 20:58:17,260 filetools.py:555 WARNING Download of https://pypi.%(namelower)s.org/packages/source/s/setuptools/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, done trying
== 2018-10-12 20:58:17,260 easyblock.py:728 DEBUG Trying to download file setuptools-39.2.0.zip from http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip ...
== 2018-10-12 20:58:17,260 filetools.py:484 DEBUG Trying to download setuptools-39.2.0.zip from http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip
== 2018-10-12 20:58:17,261 filetools.py:491 DEBUG Using timeout of 10 seconds for initiating download
== 2018-10-12 20:58:17,261 filetools.py:1212 DEBUG Not creating existing path /home/zao/.local/eb-develop/sources/p/Python/extensions
== 2018-10-12 20:58:17,471 filetools.py:529 WARNING URL http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip was not found (HTTP response code 404), not trying again
== 2018-10-12 20:58:17,471 filetools.py:555 WARNING Download of http://www.python.org/ftp/python/2.7.15/setuptools-39.2.0.zip to /home/zao/.local/eb-develop/sources/p/Python/extensions/setuptools-39.2.0.zip failed, done trying
== 2018-10-12 20:58:17,496 build_log.py:158 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Couldn't find file setuptools-39.2.0.zip anywhere, and downloading it didn't work either...
```

**Comment by boegel** [comment]
@zao That's an entirely different problem, please open a separate issue on that.

**Comment by boegel** [comment]
fixed with #2617, to be included in upcoming EasyBuild v3.7.1...

**Comment by pescobar** [comment]
thank you for the quick fix!
